django/contrib/admindocs/views.py

```diff
@@ -404,21 +404,25 @@ def get_context_data(self, **kwargs):
             # Non-trivial TEMPLATES settings aren't supported (#24125).
             pass
         else:
-            # This doesn't account for template loaders (#24128).
-            for index, directory in enumerate(default_engine.dirs):
+            directories = list(default_engine.dirs)
+            for loader in default_engine.template_loaders:
+                if hasattr(loader, "get_dirs"):
+                    directories.extend(loader.get_dirs())
```

Comment:
Is it worth going back and forward on the list of directories? We need to evaluate the entire list, so we can do this at the beginning and avoid doing this twice, for example:
```diff
diff --git a/django/contrib/admindocs/views.py b/django/contrib/admindocs/views.py
index bd9cd952df..f4c5509390 100644
--- a/django/contrib/admindocs/views.py
+++ b/django/contrib/admindocs/views.py
@@ -407,13 +407,11 @@ class TemplateDetailView(BaseAdminDocsView):
             directories = list(default_engine.dirs)
             for loader in default_engine.template_loaders:
                 if hasattr(loader, "get_dirs"):
-                    directories.extend(loader.get_dirs())
-            # Go backwards over directories and ignore duplicates on the way. Template
-            # uses dictsort to show directories in right order and with right index.
-            index = 0
-            while directories:
-                if not (directory := directories.pop()) or directory in directories:
-                    continue
+                    for dir_ in loader.get_dirs():
+                        if dir_ not in directories:
+                            directories.append(dir_)
+
+            for index, directory in enumerate(directories):
                 template_file = Path(safe_join(directory, template))
                 templates.append(
                     {
@@ -422,7 +420,6 @@ class TemplateDetailView(BaseAdminDocsView):
                         "order": index,
                     }
                 )
-                index -= 1
         return super().get_context_data(
             **{
                 **kwargs,

```

Reply:
True, that's better.